asset/swift_delete.go

```diff
@@ -0,0 +1,93 @@
+package asset
+
+import (
+	"errors"
+	"github.com/cozy/cozy-apps-registry/utils"
+	"github.com/hashicorp/go-multierror"
+	"github.com/ncw/swift"
+	"time"
+)
+
+// maxNbFilesToDelete is the maximal number of files that we will try to delete
+// in a single call to swift.
+const maxNbFilesToDelete = 8000
+
+// maxSimultaneousCalls is the maximal number of simultaneous calls to Swift to
+// delete files in the same container.
+const maxSimultaneousCalls = 8
+
+var errFailFast = errors.New("fail fast")
+
+// DeleteContainer removes all the files inside the given container, and then
+// deletes it.
+func DeleteContainer(c *swift.Connection, container string) error {
```

Comment:
What's the articulation of this function with `asset/swift.go:EmptyThenDeleteContainer()` ?
seems like code duplication.

Reply:
Was previous code, quite buggy because don't take care of bulk limit.
Not used anymore, can be removed now.

Reply:
Yes, we should use this version from the stack, and remove `asset/swift.go:EmptyThenDeleteContainer()`!

Reply:
Done.